sycl/test/tools/abi_check.dump

```diff
@@ -0,0 +1,8 @@
+# RUN: not env LLVM_BIN_PATH=%llvm_build_bin_dir python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %llvm_build_lib_dir/libsycl.so | FileCheck %s
```

Comment:
Thanks. I think the basic set of test cases should include all the 5 possible outcomes you listed abive - new symbol (done), missing symbol, changed size, changed alignment and no changes.

Reply:
@kbobrovs we can consider missing symbols also covered. This test compares dump (this file) against libsycl.so. The python script, however, does not check symbol size. There's rather a simple cpp file with a bunch of static_asserts for that, I'm not sure we need testing for that. And neither of ABI tests currently checks for alignment, so there's nothing to verify at the point.

Reply:
I see. Then the only case uncovered for the python script is "no changes".

Does it make sense then to have e.g. symbol_size_neg.cpp which would contain something like
  `check_size<accessor_t, 31>();`
and marked XFAIL: *
?

Reply:
The no changes case is a bit harder to implement. We would need a binary to test on.

As for testing check_size, i see no reason for that (unless you suspect a bug in `static_assert` or `sizeof`).